modules/hdf4_handler/HDFDMRArray_EOS2LL.cc

```diff
@@ -0,0 +1,283 @@
+/////////////////////////////////////////////////////////////////////////////
+// This file is part of the hdf4 data handler for the OPeNDAP data server.
+// It retrieves HDF-EOS2 latitude/longtitude  values for a direct DMR-mapping DAP4 response.
+// Each SDS will be mapped to a DAP variable.
+
+//  Authors:   Kent Yang <myang6@hdfgroup.org>
+// Copyright (c) The HDF Group
+/////////////////////////////////////////////////////////////////////////////
+
+
+#include "HDFDMRArray_EOS2LL.h"
+#include "hdf.h"
+#include "mfhdf.h"
+#include "HdfEosDef.h"
+#include <iostream>
+#include <sstream>
+#include <libdap/debug.h>
+#include <libdap/InternalErr.h>
+#include <BESDebug.h>
+
+using namespace std;
+using namespace libdap;
+
+
+bool
+HDFDMRArray_EOS2LL::read ()
+{
+
+    BESDEBUG("h4","Coming to HDFDMRArray_EOS2LL read "<<endl);
+    if (length() == 0)
+        return true; 
+
+    // Declaration of offset,count and step
+    vector<int>offset;
+    offset.resize(ll_rank);
+    vector<int>count;
+    count.resize(ll_rank);
+    vector<int>step;
+    step.resize(ll_rank);
+
+    string err_msg;
+
+    // Obtain offset,step and count from the client expression constraint
+    int nelms = format_constraint(offset.data(),step.data(),count.data());
+
+    int32 gridfd = GDopen(const_cast < char *>(filename.c_str()), DFACC_READ);
+    if (gridfd <0) {
+        ostringstream eherr;
+        err_msg = "HDF-EOS: GDopen failed";
+        throw InternalErr (__FILE__, __LINE__, err_msg);
```

Comment:
These would be better as BESInternalError() exceptions. Lets not make that change with this PR, but make a new ticket for that change since I think there will be many of these call to update. Also, it's not really anything wrong with this PR but instead it is an improvement to the handler.

Reply:
Will do.